oqs/oqs.py

```diff
@@ -1,29 +1,43 @@
 """
-Open Quantum Safe (OQS) Python wrapper for liboqs
+Open Quantum Safe (OQS) Python wrapper for liboqs.
 
 The liboqs project provides post-quantum public key cryptography algorithms:
 https://github.com/open-quantum-safe/liboqs
 
 This module provides a Python 3 interface to liboqs.
 """
 
+from __future__ import annotations
+
 import ctypes as ct  # to call native
 import ctypes.util as ctu
 import importlib.metadata  # to determine module version at runtime
-import os  # to run OS commands (install liboqs on demand if not found)
+import logging
 import platform  # to learn the OS we're on
+import subprocess
 import sys
 import tempfile  # to install liboqs on demand
 import time
 import warnings
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, ClassVar, Final, TypeVar, cast
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from types import TracebackType
+
+TKeyEncapsulation = TypeVar("TKeyEncapsulation", bound="KeyEncapsulation")
+TSignature = TypeVar("TSignature", bound="Signature")
 
+logger = logging.getLogger(__name__)
 
-def oqs_python_version():
+
+def oqs_python_version() -> str | None:
```

Comment:
Hello,
Using pipe operator as shorthand for Union will not work on Python 3.9 (This feature was introduced in Python3.10.